File: server/src/main/java/org/codi/catan/model/game/Phase.java
/*
 * @author the-codinator
 * created on 2020/5/28
 */

package org.codi.catan.model.game;

import lombok.AllArgsConstructor;
import lombok.Getter;

@AllArgsConstructor
@Getter
public enum Phase {
    setup1(true),
    setup2(true),
    gameplay,
    thief,
    end;

    Phase() {
        this(false);
    }

    private final boolean setupPhase;
}

File: server-azure/src/model/game/phase.ts
export enum Phase {
  setup1 = 'setup1',
  setup2 = 'setup2',
  gameplay = 'gameplay',
  thief = 'thief',
  end = 'end',
}

export function isSetupPhase(phase: Phase): boolean {
  return phase === Phase.setup1 || phase === Phase.setup2;
}
